File: C_Assignment/Assignment_44/programAQ2.c
///////////////////////////////////////////////////////////
//
//  File name : Assignement44Question2.c
//  Description : Write a program which accept matrix and one number from user and return
//                frequency of that number.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*

Input :
Number : 9
3 2 5 9
4 3 2 2
8 4 1 5
3 9 7 5

Output : 9
*/

#include<stdio.h>

int CountFrequency(int Arr[][4],int iRow,int iCol,int iNo)
{
  int i = 0, j = 0,iCount = 0;

    for(i = 0;i < iRow ;i++)
    {
        for(j = 0;j < iCol;j++)
        {
            if(Arr[i][j] == iNo)
            {
                iCount++;
            }
        }
    }

    return iCount;
}

int main()
{
    int iRow = 0,iCol = 0,i = 0,j = 0;
    int Arr[4][4];
    int iRet = 0,iNo = 0;
    
    printf("Enter number of rows :\n");
    scanf("%d",&iRow);

    printf("Enter number of columns :\n");
    scanf("%d",&iCol);

    printf("Enter matrix elements :\n");
    for(i = 0;i < iRow;i++)
    {
        for(j = 0;j < iCol;j++)
        {
            scanf("%d",&Arr[i][j]);
        }
    }

    printf("Enter number to search : \n");
    scanf("%d",&iNo);
    
    iRet = CountFrequency(Arr,iRow,iCol,iNo);

    printf("Frequency of %d is : %d\n",iNo,iRet);

    return 0;
}

File: Java_Assignment/Assignment_60/programAQ2.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement60Question2.java
//  Description : Write Java program to multiply two matrices
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

import java.util.Scanner;

class MatrixMultiplication
{
    public static void main(String args[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter rows and columns of first matrix: ");
        int row1 = sobj.nextInt();
        int col1 = sobj.nextInt();

        // Input rows and columns for second matrix
        System.out.print("Enter rows and columns of second matrix: ");
        int row2 = sobj.nextInt();
        int col2 = sobj.nextInt();

        // Check if multiplication is possible
        if (col1 != row2) {
            System.out.println("Matrix multiplication not possible.");
            return;
        }

        int A[][] = new int[row1][col1];
        int B[][] = new int[row2][col2];
        int result[][] = new int[row1][col2];

        // Input first matrix
        System.out.println("Enter elements of first matrix:");
        for (int i = 0; i < row1; i++) {
            for (int j = 0; j < col1; j++) {
                A[i][j] = sobj.nextInt();
            }
        }

        // Input second matrix
        System.out.println("Enter elements of second matrix:");
        for (int i = 0; i < row2; i++) {
            for (int j = 0; j < col2; j++) {
                B[i][j] = sobj.nextInt();
            }
        }

        // Matrix multiplication logic
        for (int i = 0; i < row1; i++) {
            for (int j = 0; j < col2; j++) {
                result[i][j] = 0;
                for (int k = 0; k < col1; k++) {
                    result[i][j] += A[i][k] * B[k][j];
                }
            }
        }

        // Display result
        System.out.println("Resultant Matrix:");
        for (int i = 0; i < row1; i++) {
            for (int j = 0; j < col2; j++) {
                System.out.print(result[i][j] + " ");
            }
            System.out.println();
        }

        sobj.close();
    }
}